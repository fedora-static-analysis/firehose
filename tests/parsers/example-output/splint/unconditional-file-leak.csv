Warning, Flag Code, Flag Name, Priority, File, Line, Column, Warning Text, Additional Text
1,136,internalglobs,1,examples/unconditional-file-leak.c,7,7,"Called procedure fopen may access file system state, but globals list does not include globals fileSystem","A called function uses internal state, but the globals list for the function being checked does not include internalState"
2,126,modfilesys,1,examples/unconditional-file-leak.c,7,7,"Undocumented modification of file system state possible from call to fopen: fopen(filename, "w")","report undocumented file system modifications (applies to unspecified functions if modnomods is set)"
3,136,internalglobs,1,examples/unconditional-file-leak.c,9,5,"Called procedure fprintf may access file system state, but globals list does not include globals fileSystem","A called function uses internal state, but the globals list for the function being checked does not include internalState"
4,126,modfilesys,1,examples/unconditional-file-leak.c,9,5,"Undocumented modification of file system state possible from call to fprintf: fprintf(f, "%i: %i", i, i * i)","report undocumented file system modifications (applies to unspecified functions if modnomods is set)"
5,2,nullpass,1,examples/unconditional-file-leak.c,9,14,"Possibly null storage f passed as non-null param: fprintf (f, ...)","A possibly null pointer is passed as a parameter corresponding to a formal parameter with no /*@null@*/ annotation.  If NULL may be used for this parameter, add a /*@null@*/ annotation to the function parameter declaration."
6,182,forblock,1,examples/unconditional-file-leak.c,9,5,"Body of for statement is not a block: fprintf(f, "%i: %i", i, i * i);","Loop body is a single statement, not a compound block."
7,300,fcnuse,1,examples/unconditional-file-leak.c,3,6,"Function test declared but not used","A function is declared but not used. Use /*@unused@*/ in front of function header to suppress message."
8,295,exportheader,1,examples/unconditional-file-leak.c,3,6,"Function test exported but not declared in header file","A declaration is exported, but does not appear in a header file."
