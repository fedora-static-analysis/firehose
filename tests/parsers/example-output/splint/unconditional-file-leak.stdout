examples/unconditional-file-leak.c: (in function test)
examples/unconditional-file-leak.c:7:7:
    Called procedure fopen may access file system state, but globals list does
    not include globals fileSystem
  A called function uses internal state, but the globals list for the function
  being checked does not include internalState (Use -internalglobs to inhibit
  warning)
examples/unconditional-file-leak.c:7:7:
    Undocumented modification of file system state possible from call to fopen:
    fopen(filename, "w")
  report undocumented file system modifications (applies to unspecified
  functions if modnomods is set) (Use -modfilesys to inhibit warning)
examples/unconditional-file-leak.c:9:5:
    Called procedure fprintf may access file system state, but globals list
    does not include globals fileSystem
examples/unconditional-file-leak.c:9:5:
    Undocumented modification of file system state possible from call to
    fprintf: fprintf(f, "%i: %i", i, i * i)
examples/unconditional-file-leak.c:9:14:
    Possibly null storage f passed as non-null param: fprintf (f, ...)
  A possibly null pointer is passed as a parameter corresponding to a formal
  parameter with no /*@null@*/ annotation.  If NULL may be used for this
  parameter, add a /*@null@*/ annotation to the function parameter declaration.
  (Use -nullpass to inhibit warning)
   examples/unconditional-file-leak.c:7:7: Storage f may become null
examples/unconditional-file-leak.c:9:5: Body of for statement is not a block:
                                           fprintf(f, "%i: %i", i, i * i);
  Loop body is a single statement, not a compound block. (Use -forblock to
  inhibit warning)
examples/unconditional-file-leak.c:3:6: Function test declared but not used
  A function is declared but not used. Use /*@unused@*/ in front of function
  header to suppress message. (Use -fcnuse to inhibit warning)
   examples/unconditional-file-leak.c:10:1: Definition of test
examples/unconditional-file-leak.c:3:6:
    Function test exported but not declared in header file
  A declaration is exported, but does not appear in a header file. (Use
  -exportheader to inhibit warning)
   examples/unconditional-file-leak.c:10:1: Definition of test

